File: EHaskins.Frc.Dashboard.Old/DS Shared/Configuration.vb
Public Module Configuration
    Const DEFAULT_USER_CONTROL_DATA_SIZE As Integer = 936
    Const DEFAULT_INVALID_PACKET_SAFETY_LEVEL As Integer = 25
    Const DS_TO_ROBOT_LOCAL_PORT As Integer = 1115
    Const DS_TO_ROBOT_REMOTE_PORT As Integer = 1110
    Const ROBOT_TO_DS_LOCAL_PORT As Integer = 1150
    Const ROBOT_TO_DS_REMOTE_PORT As Integer = 1026

    Dim _userControlDataSize As Integer = DEFAULT_USER_CONTROL_DATA_SIZE
    Private _invalidPacketCountSafety As Integer = DEFAULT_INVALID_PACKET_SAFETY_LEVEL

    Private _dsToRobotRemotePortNumber As Integer = DS_TO_ROBOT_REMOTE_PORT
    Private _dsToRobotLocalPortNumber As Integer = DS_TO_ROBOT_REMOTE_PORT
    Private _robotToDsRemotePortNumber As Integer = ROBOT_TO_DS_LOCAL_PORT
    Private _robotToDsLocalPortNumber As Integer = ROBOT_TO_DS_LOCAL_PORT

    Public Property DsToRobotLocalPortNumber() As Integer
        Get
            Return _dsToRobotLocalPortNumber
        End Get
        Set(ByVal Value As Integer)
            _dsToRobotLocalPortNumber = Value
        End Set
    End Property
    Public Property DsToRobotRemotePortNumber() As Integer
        Get
            Return _dsToRobotRemotePortNumber
        End Get
        Set(ByVal Value As Integer)
            _dsToRobotRemotePortNumber = Value
        End Set
    End Property
    Public Property RobotToDsLocalPortNumber() As Integer
        Get
            Return _robotToDsLocalPortNumber
        End Get
        Set(ByVal Value As Integer)
            _robotToDsLocalPortNumber = Value
        End Set
    End Property
    Public Property RobotToDsRemotePortNumber() As Integer
        Get
            Return _robotToDsRemotePortNumber
        End Get
        Set(ByVal Value As Integer)
            _robotToDsRemotePortNumber = Value
        End Set
    End Property

    Public ReadOnly Property InvalidPacketCountSafety() As Integer
        Get
            Return _invalidPacketCountSafety
        End Get
    End Property

    Public Property UserControlDataSize() As Integer
        Get
            Return _userControlDataSize
        End Get
        Set(ByVal value As Integer)
            _userControlDataSize = value
        End Set
    End Property

End Module

File: EHaskins.Frc.Communication.Shared/Configuration.cs
using System;

namespace EHaskins.Frc.Communication
{
    public static class Configuration
    {
        const int DEFAULT_USER_CONTROL_DATA_SIZE = 936;
        const int DEFAULT_USER_STATUS_DATA_SIZE = 984;
        const int DEFAULT_INVALID_PACKET_SAFETY_LEVEL = 500;
        const int DS_TO_ROBOT_REMOTE_PORT = 1110;
        const int ROBOT_TO_DS_LOCAL_PORT = 1150;
        static int _userControlDataSize = DEFAULT_USER_CONTROL_DATA_SIZE;
        private static int _userStatusDataSize = DEFAULT_USER_STATUS_DATA_SIZE;
        private static int _invalidPacketCountSafety = DEFAULT_INVALID_PACKET_SAFETY_LEVEL;
        private static int _dsToRobotRemotePortNumber = DS_TO_ROBOT_REMOTE_PORT;
        private static int _dsToRobotLocalPortNumber = DS_TO_ROBOT_REMOTE_PORT;
        private static int _robotToDsRemotePortNumber = ROBOT_TO_DS_LOCAL_PORT;

        private static int _robotToDsLocalPortNumber = ROBOT_TO_DS_LOCAL_PORT;

        //public static int DsToRobotLocalPortNumber
        //{
        //    get { return _dsToRobotLocalPortNumber; }
        //    set { _dsToRobotLocalPortNumber = value; }
        //}
        public static int DsToRobotDestinationPortNumber
        {
            get { return _dsToRobotRemotePortNumber; }
            set { _dsToRobotRemotePortNumber = value; }
        }
        //public static int RobotToDsLocalPortNumber
        //{
        //    get { return _robotToDsLocalPortNumber; }
        //    set { _robotToDsLocalPortNumber = value; }
        //}
        public static int RobotToDsDestinationPortNumber
        {
            get { return _robotToDsRemotePortNumber; }
            set { _robotToDsRemotePortNumber = value; }
        }

        public static int InvalidPacketCountSafety
        {
            get { return _invalidPacketCountSafety; }
        }

        public static int UserControlDataSize
        {
            get { return _userControlDataSize; }
            set { _userControlDataSize = value; }
        }
        public static int UserStatusDataSize
        {
            get { return _userStatusDataSize; }
            set
            {
                _userStatusDataSize = value;
            }
        }
        

    }
}
